src/python/pants/pantsd/service/store_gc_service.py

```diff
@@ -0,0 +1,62 @@
+# coding=utf-8
+# Copyright 2018 Pants project contributors (see CONTRIBUTORS.md).
+# Licensed under the Apache License, Version 2.0 (see LICENSE).
+
+from __future__ import (absolute_import, division, generators, nested_scopes, print_function,
+                        unicode_literals, with_statement)
+
+import logging
+import threading
+import time
+
+from pants.pantsd.service.pants_service import PantsService
+
+class StoreGCService(PantsService):
+  """Store Garbage Collection Service.
+
+  This service both ensures that in-use files continue to be present in the engine's Store, and
+  performs occasional garbage collection to bound the size of the engine's Store.
+  """
+
+  def __init__(self, scheduler):
+    super(StoreGCService, self).__init__()
+    self._scheduler = scheduler
+    self._logger = logging.getLogger(__name__)
+    self._lease_extension_thread = None
+    self._garbage_collection_thread = None
+
+  def setup(self, lifecycle_lock, fork_lock):
+    super(StoreGCService, self).setup(lifecycle_lock, fork_lock)
+
+  def _extend_lease(self):
+    while True:
+      self._logger.debug("Extending leases")
+      self._scheduler.lease_files_in_graph()
+      self._logger.debug("Done extending leases")
+      time.sleep(30 * 60)
```

Comment:
I'm fine with the intervals being hardcoded for now, but they should be in constants.